packages/perseus/src/widget-type-utils.test.ts

```diff
@@ -0,0 +1,271 @@
+    describe("getWidgetTypeByWidgetKey", () => {
+        it("returns widget type when found", () => {
+            // Assemble
+            const widgetId = "dont-look-for-type-in-key";
+            const widgetMap = {
+                [widgetId]: generateTestRadioWidget(),
+            };
+
+            // Act
+            const widgetType = getWidgetTypeByWidgetId(widgetId, widgetMap);
+
+            // Assert
+            expect(widgetType).toBe("radio");
+        });
+
+        it("returns null when not found", () => {
+            // Assemble
+            const widgetId = "dont-look-for-type-in-key";
+            const widgetMap = {};
+
+            // Act
+            const widgetType = getWidgetTypeByWidgetId(widgetId, widgetMap);
+
+            // Assert
+            expect(widgetType).toBeNull();
+        });
+    });
+
+    describe("contentHasWidgetType", () => {
+        it("returns true when found", () => {
+            // Assemble
+            const type = "radio";
+            const widgetId = "dont-look-for-type-in-key";
+            const content = `[[☃ ${widgetId}]]`;
+            const widgetMap = {
+                [widgetId]: generateTestRadioWidget(),
+            };
+
+            // Act
+            const result = contentHasWidgetType(type, content, widgetMap);
+
+            // Assert
+            expect(result).toBeTrue();
+        });
+
+        it("returns false when found", () => {
+            // Assemble
+            const type = "radio";
+            const widgetId = "dont-look-for-type-in-key";
+            const content = "hello world";
+            const widgetMap = {
+                [widgetId]: generateTestRadioWidget(),
+            };
+
+            // Act
+            const result = contentHasWidgetType(type, content, widgetMap);
+
+            // Assert
+            expect(result).toBeFalse();
+        });
+    });
+
+    describe("getWidgetIdsFromContent", () => {
+        it("finds single widgetId", () => {
+            // Assemble
+            const widgetId = "dont-look-for-type-in-key";
+            const content = `[[☃ ${widgetId}]]`;
+
+            // Act
+            const result = getWidgetIdsFromContent(content);
+
+            // Assert
+            expect(result.length).toBe(1);
+            expect(result[0]).toBe(widgetId);
+        });
+
+        it("finds multiple widgetIds", () => {
+            // Assemble
+            const widgetId1 = "dont-look-for-type-in-key";
+            const widgetId2 = "this-isnt-a-widget-type";
+            const content = `[[☃ ${widgetId1}]] [[☃ ${widgetId2}]]`;
+
+            // Act
+            const result = getWidgetIdsFromContent(content);
+
+            // Assert
+            expect(result.length).toBe(2);
+            expect(result[0]).toBe(widgetId1);
+            expect(result[1]).toBe(widgetId2);
+        });
+
+        it("returns an empty array when nothing is found", () => {
+            // Assemble
+            const content = `hello world`;
```

Comment:
Suggest
```suggestion
            const content = "hello world";
```
Suggestion, nb:  I don't think it matters, but should this use double quotes when there is no interpolation?